packages/field-base/test/validate-mixin.test.js

```diff
@@ -185,5 +185,21 @@ describe('validate-mixin', () => {
       element.validate();
       expect(element.invalid).to.be.false;
     });
+
+    it('should override explicitly set invalid when required is set', () => {
+      element.invalid = true;
+      element.value = 'foo';
+      element.required = true;
+      expect(element.invalid).to.be.false;
+    });
```

Comment:
Would it be reasonable to add a similar test for [`PatternMixin`](https://github.com/vaadin/web-components/pull/2350/files#diff-7492a4a8b2132bcf8316d7f61193670ee852e32eba29a7cf32fd7a3a6070b406R189-R194)? And for [`CharLengthMixin`](https://github.com/vaadin/web-components/pull/2350/files#diff-dbe6d41254bf00d1719f2bf6b68d31b62568763776226a3e373804f28fea9e4cR67) as well?

Reply:
Created a new mixin `InputConstraintsMixin` which contains this test.